### PR TITLE
Add hwcservice_test

### PR DESCRIPTION
**Pull request by XiaosongWei**
Change-Id: Ie4be502b2b9ddec1867ad0b9fcd6e99648a022a0
Tracked-On:
Signed-off-by: Xiaosong Wei <xiaosong.wei@intel.com>

### CODE REVIEW

**Comment by XiaosongWei** [comment]
@kalyankondapally 
@feiwan1 a simple native test app, you could extend that to verify HSBC feature.

**Comment by kalyankondapally** [comment]
We already have one here: https://github.com/intel/IA-Hardware-Composer/tree/master/os/android/libhwcservice/utils  we can re-use that ?

**Comment by XiaosongWei** [comment]
I prefer to remove the utils one as it's not easy to be extended and under a bad naming folder.

File: os/android/libhwcservice/test/hwcservice_test.cpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      case 'h':
+      default:
+        usage();
+    }
+  }
+  argc -= optind;
+  argv += optind;
+
+  // Initialize ProcessState with /dev/vndbinder as HwcService is in vndbinder context
+  android::ProcessState::initWithDriver("/dev/vndbinder");
```
Comment (kalyankondapally) [inline os/android/libhwcservice/test/hwcservice_test.cpp]: Move this under USE_PROCESS_STATE checks

**Comment by kalyankondapally** [comment]
Than please update the patch to include deleting that file.

**Comment by kalyankondapally** [comment]
Please check the commit format here:'https://github.com/intel/IA-Hardware-Composer/wiki/Contributions

File: os/android/libhwcservice/test/hwcservice_test.cpp (hunk 1 above)
Comment (XiaosongWei) [reply os/android/libhwcservice/test/hwcservice_test.cpp]: Where is this macro defined? Is it defined by default under O?

Comment (kalyankondapally) [reply os/android/libhwcservice/test/hwcservice_test.cpp]: Ya, https://github.com/intel/IA-Hardware-Composer/blob/master/Android.common.mk#L126

**Comment by kalyankondapally** [comment]
lgtm

**Pull request closed without merging**